### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by DisasterMoo**
- Improved performance over small ore generation
- Added a function in `VeinRegistry` so addons can register veins without overriding config file.

### CODE REVIEW

**Comment by alcatrazEscapee** [comment]
No. Addons should not force enable veins via code. That defeats the entire purpose of making it json configurable. They should provide resources the same way, i.e. by manually copying a default json file. The required change here would be to make TFC search for all files in the default folder, not just the single specified one.

**Comment by DisasterMoo** [comment]
My intention is exactly that, not forcing via code but adding ores via json without overriding the base TFC one.

**Comment by alcatrazEscapee** [comment]
This is not a good way to expose that hook to addons. Instead, if this is necessary to avoid overwriting TFC's default ore gen, TFC should just scan for all files in the `/config/tfc/` folder, and the addon should just copy their files there.

There's no need to duplicate json reading from multiple mods. And there's no reason to expose a hook that can be used to force veins to exist. For the same reason that forge doesn't expose direct hooks to register models, blockstates, or recipes.

**Comment by DisasterMoo** [comment]
> TFC should just scan for all files in the /config/tfc/ folder
But that's exactly what i just did in the last commit.

**Comment by alcatrazEscapee** [comment]
Ah, I didn't see that last change before I commented, sorry.

**Comment by aldo1028** [comment]
any chance of this getting added soon Alc?

File: src/main/java/net/dries007/tfc/world/classic/worldgen/WorldGenLooseRocks.java [hunk 1]
```diff
@@ -68,20 +68,9 @@ public void generate(Random random, int chunkX, int chunkZ, World world, IChunkG
                 {
                     veins.removeIf(v -> {
                         if (!v.type.hasLooseRocks()) return true;
-
```
Comment (alcatrazEscapee) [inline src/main/java/net/dries007/tfc/world/classic/worldgen/WorldGenLooseRocks.java]: If we're going to remove the full scan check for ore blocks, we should introduce a better check for vein proximity:
 - In 1.7.10, it was pointed out to me that the maximum depth scanned was much smaller than we have here, it should probably be adjusted to match. We can do a check of the "center" of the vein, and use the vertical size + vein size modifier to get an approximate value for the highest ore block.
 - The horizontal detection is (at the moment) chunk restricted. Ideally, the veins will still use position / distances to check validity, and the chunk data ore only becomes a sanity check that any ores actually generated.

File: src/main/java/net/dries007/tfc/world/classic/worldgen/vein/VeinRegistry.java [hunk 2]
```diff
@@ -61,53 +61,73 @@ public void preInit(File dir)
...
             try
             {
                 weightedVeinTypes.clear();
                 veinTypeRegistry.clear();
-                Map<String, VeinType> values = GSON.fromJson(worldGenData, new TypeToken<Map<String, VeinType>>() {}.getType());
-                values.forEach((name, veinType) -> {
-                    veinType.setRegistryName(name);
-                    veinTypeRegistry.put(name, veinType);
-                    weightedVeinTypes.add(veinType.weight, veinType);
+                worldGenData.forEach(data -> {
```
Comment (alcatrazEscapee) [inline src/main/java/net/dries007/tfc/world/classic/worldgen/vein/VeinRegistry.java]: The try / catch block here should be nested as close to the GSON call as possible, so if one file has a syntax error, it doesn't block all files from processing.

**Comment by DisasterMoo** [comment]
Since I feel this needs some screenshot samples:

image

image

Small cassiterite generated about 1 chunk after the ones with that vein.

File: src/main/java/net/dries007/tfc/world/classic/worldgen/WorldGenLooseRocks.java (hunk 1 above)
Comment (DisasterMoo) [reply src/main/java/net/dries007/tfc/world/classic/worldgen/WorldGenLooseRocks.java]: To pin this down, i'm commenting the line i found in classic:
https://github.com/TerraFirmaCraft/TFC-1.7.10/blob/9979960a7c13c19e4e6d018aa774c8a707c85592/src/Common/com/bioxx/tfc/WorldGen/Generators/WorldGenLooseRocks.java#L61

**Pull request merged**